phpBB/includes/ucp/ucp_attachments.php

```diff
@@ -228,4 +233,29 @@ function main($id, $mode)
 		$this->tpl_name = 'ucp_attachments';
 		$this->page_title = 'UCP_ATTACHMENTS';
 	}
+
+	/**
+	 * Check if the user can delete the file
+	 *
+	 * @param array $row
+	 *
+	 * @return bool True if user can delete the file, false if not
+	 */
+	private function can_delete_file(array $row): bool
+	{
+		global $auth, $config;
+
+		if ($row['in_message'])
+		{
+			return ($row['message_time'] > time() - ($config['pm_edit_time'] * 60) || !$config['pm_edit_time']) && $row['folder_id'] == PRIVMSGS_OUTBOX && $auth->acl_get('u_pm_edit');
```

Comment:
You should add more parenthesis around `($row['message_time'] > time() - ($config['pm_edit_time'] * 60) || !$config['pm_edit_time'])`
The evaluation order is not explicit (`>` vs `-` vs `||`)

Reply:
I think it's definitely a bit clearer now.